src/graphapi/find_engine.cpp

```diff
@@ -36,12 +36,148 @@
+
+        return graph_gen->graph();
+    }
+
+    static bool IsBiasNode(OperationPointwise* addNode)
+    {
+        auto& lengthsToCheck  = addNode->getX()->isVirtual() ? addNode->getB()->GetLengths()
+                                                             : addNode->getX()->GetLengths();
+        int notOneOrZeroCount = 0;
+        for(auto& length : lengthsToCheck)
+        {
+            if(length > size_t{1})
+            {
+                notOneOrZeroCount++;
+            }
+        }
+
+        // Assume bias node when at most one dimension is larger than 1.
+        return notOneOrZeroCount <= 1;
+    }
+
+public:
+    static std::unique_ptr<GraphPatternMatcher> Make()
+    {
+        return std::make_unique<ConvBiasResAddActive_Fwd_Pattern>();
+    }
+
+    std::string_view name() const final
+    {
+        static const char* n = "convbiasresaddactivation_fwd";
+        return n;
+    }
+
+    bool matches(const OpGraph* graph_ptr) const final
+    {
+        assert(graph_ptr);
+        return isIsomorphic(*graph_ptr, getPatternGraph());
+    }
+
+    std::vector<Engine> getEngines(OpGraph* graph_ptr) const override
+    {
+        assert(graph_ptr);
+        assert(matches(graph_ptr));
+        auto& graph = *graph_ptr;
+
+        auto* conv = dynamic_cast<OperationConvolutionForward*>(
+            graph.findOutNeighByName(graph.getSourceNode(), "OP_CONVOLUTION_FORWARD"));
+        assert(conv);
+
+        std::size_t in_c  = conv->getX()->GetLengths()[1];
+        std::size_t wei_c = conv->getW()->GetLengths()[1];
+
+        if(wei_c == std::size_t{0})
+        {
+            MIOPEN_THROW(
+                miopenStatusBadParm,
+                "invalid weight tensor provided for graph matching ConvBiasResAddActive pattern");
+        }
+        else if(in_c % wei_c != std::size_t{0})
+        {
+            MIOPEN_THROW(miopenStatusBadParm,
+                         "invalid group count from input and weight tensor for graph matching "
+                         "ConvBiasResAddActive pattern");
+        }
+
+        int groupCount = in_c / wei_c;
+
+        auto* add1 =
+            dynamic_cast<OperationPointwise*>(graph.findOutNeighByName(conv, "OP_POINTWISE:ADD"));
+        assert(add1);
+
+        auto* add2 =
+            dynamic_cast<OperationPointwise*>(graph.findOutNeighByName(add1, "OP_POINTWISE:ADD"));
+        assert(add2);
+
+        bool isBiasNode1 = IsBiasNode(add1);
+        bool isBiasNode2 = IsBiasNode(add2);
+
+        if(!isBiasNode1 && !isBiasNode2)
+        {
+            MIOPEN_THROW(miopenStatusBadParm,
+                         "no bias node provided for graph matching ConvBiasResAddActive pattern");
+        }
+
+        auto* activ = dynamic_cast<OperationPointwise*>(
+            graph.findOutNeighByName(add2, "OP_POINTWISE:RELU_FWD"));
+        assert(activ);
+
+        auto [add, bias] = isBiasNode1 ? std::make_pair(add2, add1) : std::make_pair(add1, add2);
```

Comment:
I don't know if there's flexibility about the position of Bias node. Usually Bias follows Conv directly.

Reply:
I was wondering that.
I figured since they were the same node type, and that they are both adds, then it would be more flexible this way, but I could swap it to enforce bias after conv if that's better.